### PR TITLE
[Build] Upgrade Gradle to 8.2.1 and AGP to 8.1.0

### PR DESCRIPTION
**Pull request by ParaskP7**
Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

This PR upgraded Gradle to [8.2.1](https://docs.gradle.org/8.2.1/release-notes.html) and AGP to [8.0.2](https://developer.android.com/build/releases/gradle-plugin#agp-8-0-2).

FYI: The [8.1.1](https://docs.gradle.org/8.1.1/release-notes.html) Gradle upgrades had to happen in 2 parts (9848f1780f6109f723c561536653063fb44df097 and 8f176b09210f01cffd3e9da9be969c6952da6304). This is because of the breaking changes that this upgrade introduces, including the `AGP` upgrade and its associated breaking changes (namespace, build option, etc).

-----

🚫 Blocked By:

- [x] `Android Gradle plugin requires Java 17 to run. You are currently using Java 11.`

<details>
    <summary>CI Failure</summary>

```
[2023-07-12T11:41:19Z] FAILURE: Build failed with an exception.
--
  | [2023-07-12T11:41:19Z]
  | [2023-07-12T11:41:19Z] * Where:
  | [2023-07-12T11:41:19Z] Build file '/var/lib/buildkite-agent/builds/ci-android-i-00498079ce90318ff-1/automattic/stories-android/app/build.gradle' line: 2
  | [2023-07-12T11:41:19Z]
  | [2023-07-12T11:41:19Z] * What went wrong:
  | [2023-07-12T11:41:19Z] An exception occurred applying plugin request [id: 'com.android.application', version: '8.0.2']
  | [2023-07-12T11:41:19Z] > Failed to apply plugin 'com.android.internal.application'.
  | [2023-07-12T11:41:19Z]    > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
  | [2023-07-12T11:41:19Z]       Your current JDK is located in /usr/lib/jvm/java-11-openjdk-11.0.19.0.7-1.amzn2.0.1.x86_64
  | [2023-07-12T11:41:19Z]       You can try some of the following options:
  | [2023-07-12T11:41:19Z]        - changing the IDE settings.
  | [2023-07-12T11:41:19Z]        - changing the JAVA_HOME environment variable.
  | [2023-07-12T11:41:19Z]        - changing `org.gradle.java.home` in `gradle.properties`.
```

</details>

-----

Breaking Change List:

1. [Move package from android manifest to build files](https://github.com/Automattic/stories-android/commit/49fce3675ccea4dfe4c3ae5ed0dab3c1fd07b550)
2. [Enable build config build feature for stories and app](https://github.com/Automattic/stories-android/commit/f8ac2a52ce6eca7d5f1129aa693738e19d40de10)

AGP Upgrade Assistant List:

1. [Remove ignored configuration properties](https://github.com/Automattic/stories-android/commit/cddd76546c8eb62b8ffb0729bdb88180ecc0fd0b)
2. [Migrate dsl properties from lint options to lint](https://github.com/Automattic/stories-android/commit/7357c611eb7fc69f9f12c88deadc745031c0915b)

-----

## Associated clients

You can reference the `[Build] Upgrade Gradle to 8.1.1 and AGP to 8.0.2` PRs of the below associated clients for testing/verification purposes:

- [ ] WPAndroid#<PR_NUMBER>

-----

## To test

1. Verify that all the CI checks are successful.
2. Test composite builds via the associated clients apps (see above).

-----

## Merge instructions

- [x] Mark PR as `Ready for review`.
- [x] Wait for PR to get approved.
    - [x]  TODO: Test all the PRs together once done with the reviews (@oguzkocer).
- [x]  Remove `[Status] Not Ready for Merge` label.
- [x] In coordination with all the other library and client PRs, merge this PR only when every PR is ready to be merged.

### CODE REVIEW

File: mp4compose/build.gradle [hunk 1]
```diff
@@ -5,13 +5,13 @@ plugins {
 }
 
 android {
+    namespace "com.daasuu.mp4compose"
```
Comment (oguzkocer) [inline mp4compose/build.gradle:8]: This namespace was quite unexpected 😕 However, it's the same one that we have in `AndroidManifest`. 🤷‍♂️

**Review by oguzkocer — APPROVED** [review]
Looks good to me. Approving the PR pending testing.

File: mp4compose/build.gradle (hunk 1 above)
Comment (ParaskP7) [reply mp4compose/build.gradle:8]: Yea, I know, I felt the same way when I made that change... 🤷

**Comment by ParaskP7** [comment]
Awesome @oguzkocer , thanks for adding JDK17 support to this repo, updating the `publish-to-s3` Gradle plugin, then reviewing and tentatively approving the changes, pending some extra testing! 🙇 ❤️ 🚀 

FYI: I went ahead, marked this PR as `Ready for review` and removed the `[PR] Not Ready for Merge` label. The last thing for us to do is to coordinate the merge wit with all the other library and client PRs, merging this PR only when every other such PR is ready to be merged as well.

**Pull request merged**